lib/screens/lock/lock_screen.dart

```diff
@@ -0,0 +1,241 @@
+                ),
+                kSmallSpacer,
+                _buildLockApproveButton(),
+                kMediumSpacer,
+                if (_lockState == ButtonStates.hasToApprove) _buildSteps(),
+                Spacer(),
+                if ((_lockState != ButtonStates.hasToApprove && _showToast) ||
+                    (_lockState == ButtonStates.pendingApproveDividedButton &&
+                        _showToast))
+                  _buildToast()
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  CountdownTimer _buildCountDownTimer() {
+    return CountdownTimer(
+      endTime: endTime,
+      widgetBuilder: (_, time) {
+        if (time == null) {
+          return Text(
+              'Current locking period still lasts 0 Days, 0 Hours, 0 Minutes and 0 Seconds.',
+              style: MyStyles.gradientMediumTextStyle);
+        } else {
+          return Text(
+            'Current locking period still lasts ${time?.days ?? 0} Days, ${time?.hours ?? 0} Hours, ${time?.min ?? 0} Minutes and ${time?.sec ?? 0} Seconds.',
+            style: MyStyles.gradientMediumTextStyle,
+          );
+        }
+      },
+    );
+  }
+
+  Widget _buildToast() {
+    if (_lockState != ButtonStates.pendingApproveDividedButton &&
+        _lockState != ButtonStates.pendingApproveMergedButton) {
+      return _buildTransactionSuccessToast();
+    } else {
+      return _buildTransactionPending();
+    }
+  }
+
+  Toast _buildTransactionSuccessToast() {
+    return Toast(
+      label: 'Successful',
+      color: Color(MyColors.ToastGreen),
+      onPressed: () {},
+      onClosed: () {
+        setState(() {
+          _showToast = false;
+        });
+      },
+    );
+  }
+
+  Toast _buildTransactionPending() {
+    return Toast(
+      label: 'Transaction Pending',
+      color: Color(MyColors.ToastGrey),
+      onPressed: () {},
+      onClosed: () {
+        setState(() {
+          _showToast = false;
+        });
+      },
+    );
+  }
+
+  Padding _buildSteps() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 75),
+      child: Row(
+        children: [
+          Container(
+            padding: EdgeInsets.all(8),
+            child: Text(
+              '1',
+              style: MyStyles.whiteMediumTextStyle,
+            ),
+            decoration:
+                BoxDecoration(gradient: gradient, shape: BoxShape.circle),
+          ),
+          Expanded(
+              child: Container(
+                  height: 3, decoration: BoxDecoration(gradient: gradient))),
+          Container(
+            padding: EdgeInsets.all(8),
+            child: Text(
+              '2',
+              style: MyStyles.whiteMediumTextStyle,
+            ),
+            decoration: BoxDecoration(
+                color: Color(MyColors.Button_BG_Black), shape: BoxShape.circle),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget _buildLockApproveButton() {
+    return CrossFadeButton(
```

Comment:
Ich hatte mir die Implementationen gar nicht so genau angesehen und jetzt erst realisiert, dass das ja eigentlich die zwei Buttons sind, die dann zusammenschmelzen. Im Nachhinein muss ich sagen, dass das richtig genial war, wie du den strukturiert hast. Und jetzt wo wir den verallgemeinert haben, ist das noch besser. Ich find den können wir auch für unsere interne Bibliothek einbinden :) 

Auch hier nice work.

Das kommt jetzt vielleicht etwas spät, aber mir fiel grade auf, dass man den Titel noch eindeutiger machen könnte. 
- [ ] Kannst du das in CrossFadeDuoButton umbenennen?
Mit DuoButton ist klar, dass das irgendwas mit "zwei" zu tun hat :D